File: kattis/02-24-practice/B.cc
#include <bits/stdc++.h>
using namespace std;

void solve(string &oline) {
  // remove all spaces
  string line = "";
  for (char c : oline) {
    if (c != ' ') {
      line.push_back(c);
    }
  }
  for (int l = 1; l < line.size(); ++l) {
    unordered_map<string, int> freq;
    int local_max = -1;
    for (int x{}; x < line.size(); ++x) {
      string substr = line.substr(x, l);
      ++freq[substr];
      local_max = max(local_max, freq[substr]);
    }
    if (local_max <= 1)
      return;
    cout << local_max << "\n";
  }
}

int main() {
  vector<string> lines;
  string in;
  while (getline(cin, in)) {
    lines.emplace_back(in);
  }

  for (string &line : lines) {
    solve(line);
    cout << "\n";
  }

  return 0;
}

File: kattis/02-24-practice/B.py
import sys
from collections import defaultdict

lines = [line.rstrip() for line in sys.stdin.readlines()]

def solve(line):
    line = line.replace(' ', '')
    for l in range(1, len(line)):
        freq = defaultdict(lambda: int())
        local_max = -1 # !! -1 fucky?
        for (i, v) in enumerate(line):
            substr = line[i: i+l]
            freq[substr] += 1
            local_max = max(local_max, freq[substr])
        if local_max <= 1:
            return
        print(local_max)

for line in lines:

    solve(line)
    if line != lines[-1]:
        print()
